libs/zgui/src/gui.zig

```diff
@@ -3371,6 +3371,88 @@ extern fn zguiGetMouseDragDelta(button: MouseButton, lock_threshold: f32, delta:
 extern fn zguiResetMouseDragDelta(button: MouseButton) void;
 //--------------------------------------------------------------------------------------------------
 //
+// Drag and Drop
+//
+//--------------------------------------------------------------------------------------------------
+pub const DragDropFlags = packed struct(u32) {
+    source_no_preview_tooltip: bool = false,
+    source_no_disable_hover: bool = false,
+    source_no_hold_open_to_others: bool = false,
+    source_allow_null_id: bool = false,
+    source_extern: bool = false,
+    source_auto_expire_payload: bool = false,
+
+    _padding0: u4 = 0,
+
+    accept_before_delivery: bool = false,
+    accept_no_draw_default_rect: bool = false,
+    accept_no_preview_tooltip: bool = false,
+    accept_peek_only: bool = false,
+
+    _padding1: u18 = 0,
+};
+
+const Payload = extern struct {
+    data: *anyopaque = null,
+    data_size: i32 = 0,
+    source_id: u32 = 0,
+    source_parent_id: u32 = 0,
+    data_frame_count: i32 = -1,
+    data_type: [32 + 1]u8,
+    preview: bool = false,
+    delivery: bool = false,
+
+    /// `pub fn init() Payload`
+    pub const init = zguiImGuiPayload_Init;
+    extern fn zguiImGuiPayload_Init() Payload;
+
+    /// `pub fn clear(payload: *Payload) void`
+    pub const clear = zguiImGuiPayload_Clear;
+    extern fn zguiImGuiPayload_Clear(payload: *Payload) void;
+
+    /// `pub fn isDataType(payload: *const Payload, type: [*:0]const u8) bool`
+    pub const isDataType = zguiImGuiPayload_IsDataType;
+    extern fn zguiImGuiPayload_IsDataType(payload: *const Payload, type: [*:0]const u8) bool;
+
+    /// `pub fn isPreview(payload: *const Payload) bool`
+    pub const isPreview = zguiImGuiPayload_IsPreview;
+    extern fn zguiImGuiPayload_IsPreview(payload: *const Payload) bool;
+
+    /// `pub fn isDelivery(payload: *const Payload) bool;
+    pub const isDelivery = zguiImGuiPayload_IsDelivery;
+    extern fn zguiImGuiPayload_IsDelivery(payload: *const Payload) bool;
+};
+
+pub fn beginDragDropSource(flags: DragDropFlags) bool {
+    return zguiBeginDragDropSource(flags);
+}
+pub fn setDragDropPayload(payload_type: [*:0]const u8, data: *anyopaque, sz: usize, cond: Condition) bool {
+    return zguiSetDragDropPayload(payload_type, data, sz, cond);
```

Comment:
We can capitalise on Zig's slice type here, i.e.
```zig
pub fn setDragDropPayload(payload_type: [*:0]const u8, data: []const u8, cond: Condition) bool {
     return zguiSetDragDropPayload(payload_type, data.ptr, data.len, cond);
```

To call this with something other than bytes would look like:
```zig
_ = zgui.setDragDropPayload("MyPayload", std.mem.sliceAsBytes(my_payload));
```

I think this would be nice because it prevents passing the wrong size. What do you think?

Reply:
My mistake, that should of been
```zig
    return zguiSetDragDropPayload(payload_type, @ptrCast(data.ptr), data.len, cond);
```